pyseir/rt/infer_rt.py

```diff
@@ -9,42 +9,64 @@
 from scipy import stats as sps
 from matplotlib import pyplot as plt
 
+from libs import pipeline
+from libs.datasets import timeseries
 from pyseir import load_data
-from pyseir.utils import TimeseriesType, get_run_artifact_path, RunArtifact
+from pyseir.utils import TimeseriesType, RunArtifact
 from pyseir.rt.constants import InferRtConstants
 from pyseir.rt import plotting, utils
 
 rt_log = structlog.get_logger(__name__)
 
 
-def run_rt_for_fips(
-    fips: str,
+@dataclass(frozen=True)
+class RegionalInput:
+    region: pipeline.Region
+
+    _combined_data: pipeline.RegionalCombinedData
+
+    @staticmethod
+    def from_region(region: pipeline.Region) -> "RegionalInput":
+        return RegionalInput(
+            region=region, _combined_data=pipeline.RegionalCombinedData.from_region(region),
+        )
+
+    @staticmethod
+    def from_fips(fips: str) -> "RegionalInput":
+        return RegionalInput.from_region(pipeline.Region.from_fips(fips))
+
+    def get_timeseries(self) -> timeseries.TimeseriesDataset:
+        return self._combined_data.get_timeseries()
+
+
+def run_rt(
+    regional_input: RegionalInput,
     include_deaths: bool = False,
     include_testing_correction: bool = False,
     figure_collector: Optional[list] = None,
 ):
     """Entry Point for Infer Rt"""
 
-    # TODO: This fails silently if you pass it a numeric fips instead of a string
-    # assert type(fips) == str
-
     # Generate the Data Packet to Pass to RtInferenceEngine
     input_df = _generate_input_data(
-        fips=fips,
+        regional_input=regional_input,
         include_testing_correction=include_testing_correction,
         include_deaths=include_deaths,
         figure_collector=figure_collector,
     )
     if input_df.dropna().empty:
-        rt_log.warning(event="Infer Rt Skipped. No Data Passed Filter Requirements:", fips=fips)
+        rt_log.warning(
+            event="Infer Rt Skipped. No Data Passed Filter Requirements:",
+            region=regional_input.region,
+        )
         return
 
     # Save a reference to instantiated engine (eventually I want to pull out the figure
     # generation and saving so that I don't have to pass a display_name and fips into the class
     engine = RtInferenceEngine(
         data=input_df,
-        display_name=_get_display_name(fips),
-        fips=fips,
+        display_name=str(regional_input.region),
```

Comment:
this would be great to have on the `RegionalInput` object

Reply:
Done